src/ln/channel.rs

```diff
@@ -1094,26 +1097,32 @@ impl Channel {
 	pub fn accept_channel(&mut self, msg: &msgs::AcceptChannel) -> Result<(), HandleError> {
 		// Check sanity of message fields:
 		if !self.channel_outbound {
-			return Err(HandleError{err: "Got an accept_channel message from an inbound peer", action: None});
+			return Err(HandleError{err: "Got an accept_channel message from an inbound peer", action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: "".to_string() }})});
 		}
 		if self.channel_state != ChannelState::OurInitSent as u32 {
-			return Err(HandleError{err: "Got an accept_channel message at a strange time", action: None});
+			return Err(HandleError{err: "Got an accept_channel message at a strange time", action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: "".to_string() }})});
 		}
 		if msg.dust_limit_satoshis > 21000000 * 100000000 {
-			return Err(HandleError{err: "Peer never wants payout outputs?", action: None});
+			return Err(HandleError{err: "Peer never wants payout outputs?", action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: "".to_string() }})});
 		}
 		if msg.channel_reserve_satoshis > self.channel_value_satoshis {
-			return Err(HandleError{err: "Bogus channel_reserve_satoshis", action: None});
+			return Err(HandleError{err: "Bogus channel_reserve_satoshis", action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: "".to_string() }})});
+		}
+		if msg.channel_reserve_satoshis < self.our_dust_limit_satoshis {
```

Comment:
added this also

Reply:
Cool, but there should be 4 channel_reserve_satoshis checks, not just one, no? Should check our reserves against their dust_limit as well, and also for inbound channels.